core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -919,12 +919,16 @@ class GroupCoordinator(val brokerId: Int,
 
   def onCompleteJoin(group: GroupMetadata) {
     group.inLock {
-      // remove any members who haven't joined the group yet
-      group.notYetRejoinedMembers.foreach { failedMember =>
-        removeHeartbeatForLeavingMember(group, failedMember)
-        group.remove(failedMember.memberId)
-        group.removeStaticMember(failedMember.groupInstanceId)
-        // TODO: cut the socket connection to the client
+      val leaderId = group.leaderOrNull
+      if (leaderId != null) {
+        val leader = group.get(group.leaderOrNull)
+        if (!leader.isAwaitingJoin) {
```

Comment:
If I understand correctly, we are trying to ensure that we still have a leader that can do the assignment. But is it necessary to actually remove the leader from the group? It makes the behavior a little inconsistent. Could we just change the leader?